Assets/_Application/_Twin/UI/Loader/AsyncOperationLoadingScreen.cs

```diff
@@ -0,0 +1,56 @@
+using DG.Tweening;
+using TMPro;
+using UnityEngine;
+using UnityEngine.ResourceManagement.AsyncOperations;
+
+namespace Netherlands3D.Twin.UI.Loader
+{
+    public class AsyncOperationLoadingScreen : MonoBehaviour
+    {
+        [SerializeField] private UI_ProgressIndicator progressIndicator;
+        [SerializeField] private TMP_Text labelField;
+        [SerializeField] private float fadeDuration = 0.4f;
+        [SerializeField] private string overlayCanvasTag = "ScreenOverlayCanvas";
+
+        private AsyncOperationHandle handle;
+
+        public AsyncOperationHandle Handle
+        {
+            set
+            {
+                transform.SetParent(GameObject.FindWithTag(overlayCanvasTag)?.transform, false);
+                progressIndicator.ShowProgress(0.1f);
+
+                handle = value;
+                handle.Completed += OnCompleted;
```

Comment:
als handle al een waarde had bij het binnenkomen van de set, dan moet er ook nog handle.Completed -= OnCompleted

Reply:
Dit is afgehandeld in de OnCompleted zelf, hij unregistered zichzelf na gebruik